src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1148,6 +1148,10 @@ private static void MergeHidingLookupResults(LookupResult resultHiding, LookupRe
                             // SPEC: interface type, the base types of T are the base interfaces
                             // SPEC: of T and the class type object. 
 
+                            if (hiddenContainer.Equals(hidingSym.ContainingType, TypeCompareKind.IgnoreNullableModifiersForReferenceTypes))
```

Comment:
For my eductation: can we get into a similar problem when interfaces differ by `object` and `dynamic` type parameters?

Reply:
yes, but `dynamic` is not allowed as a type argument. See `AmbigMember_DynamicDifferences` below.